lib/pbench/agent/base.py

```diff
@@ -22,16 +22,18 @@ def __init__(self, context):
         self.name = os.path.basename(sys.argv[0])
 
         env_pbench_run = os.environ.get("pbench_run")
-        self.pbench_run = pathlib.Path(
-            env_pbench_run if env_pbench_run else self.config.pbench_run
-        )
-
-        if not self.pbench_run.exists():
-            click.echo(
-                f"[ERROR] pbench run directory does not exist, {self.pbench_run}",
-                err=True,
-            )
-            click.get_current_context().exit(1)
+        if env_pbench_run:
+            self.pbench_run = pathlib.Path(env_pbench_run)
+            if not self.pbench_run.exists():
+                click.echo(
+                    f"[ERROR] the provided pbench run directory, {env_pbench_run}, does not exist",
+                    err=True,
+                )
+                click.get_current_context().exit(1)
+        else:
+            self.pbench_run = pathlib.Path(self.config.pbench_run)
+            if not self.pbench_run:
+                self.pbench_run = pathlib.Path("/var/lib/pbench-agent")
 
         # the pbench temporary directory is always relative to the $pbench_run
         # directory
```

Comment:
If `$pbench_run/tmp` already exists, the Bash code will proceed while the Python code will fail.  Is this OK/desirable?  (We can easily make the Python code match the Bash code by adding `exist_ok=True` to the `os.makedirs()` call at line 43.)

Reply:
We should make them behave the same.

Reply:
> If `$pbench_run/tmp` already exists, the Bash code will proceed while the Python code will fail

WDYM? We have this check in base.py `if not self.pbench_tmp.exists():` which should prevent python to fail if the tmp directory already exist.

I just noticed that there is one difference we need to fix between bash and python and that is `base.py` has this check `if not self.pbench_tmp.exists():` and if it returns True we create the directory using `os.makedirs(self.pbench_tmp)` which will also create the parent directories. Bash does not have this check it just proceed to create the directory with `mkdir -p` even if it exists. 

I suppose we can do 
```
try:
    os.makedirs(self.pbench_tmp, exists_ok=True)
except OSError:
    ...
``` 
instead of 
```
if not self.pbench_tmp.exists():
    try:
        os.makedirs(self.pbench_tmp)
    except OSError:
        ...
```
Is this what you mean?

Reply:
I agree that they appear to be the same.  @webbnh, can you point out where you see the difference?

Reply:
Nope -- my bad:  I somehow missed or misread the `if not self.pbench_tmp.exists()` test.

My apologies!